### PR TITLE
Completed Task 1.1, 1.2 with fix + Task 2.1-2.2 for review.

### PR DESCRIPTION
**Pull request by Valerii-Tcvetkov**

### CODE REVIEW

**Comment by Valerii-Tcvetkov** [comment]
Реализовал метод сначала для бинарной классификации, затем для многоклассовой классификации. В ходе выполнения работы определил лучшие значения k для классификации путем кросс-валидации, а также указал необходимые метрики. Финальная проверка дала точность 0.21

**Comment by Valerii-Tcvetkov** [comment]
Реализовал линейный классификатор. Лучшие результаты были достигнуты при параметрах 
num_epochs = 20
batch_size = 30
learning_rates = [1e-3, 1e-4, 1e-5]
reg_strengths = [1e-2, 1e-3, 1e-4]

Тренировка: 0.259
Тест: 0.213

File: Task1/Valerii-Tcvetkov/linear_classifer.py [hunk 1]
```diff
@@ -53,9 +92,46 @@ def softmax_with_cross_entropy(predictions, target_index):
...
+        tg_ind_flt[tuple(np.arange(0, probs.shape[0])), tuple(target_index.T[0])] = 1
+
+    # print(tg_ind_flt)
+
+    loss = -np.sum(tg_ind_flt * np.log(probs))
+
+    dprediction = probs
+    dprediction[tg_ind_flt.astype(bool)] = dprediction[tg_ind_flt.astype(bool)] - 1
+
+    # print(dprediction)
```
Comment (toshiks) [inline Task1/Valerii-Tcvetkov/linear_classifer.py]: нужно использовать написанные функции

Comment (Valerii-Tcvetkov) [reply Task1/Valerii-Tcvetkov/linear_classifer.py]: Указанный блок используется в linear_softmax для нахождения loss и dprediction.

Comment (toshiks) [reply Task1/Valerii-Tcvetkov/linear_classifer.py]: Нужно использовать softmax и cross-entropy-loss которые ты написал в этой функции

**Comment by Valerii-Tcvetkov** [comment]
Внес необходимые правки в softmax_with_cross_entropy.
При параметрах:
num_epochs = 50
batch_size = 300
learning_rates = [1e-1, 1e-2, 1e-3]
reg_strengths = [1e-1, 1e-2, 1e-3]

Тренировка: 0.25
Тест: 0.205

File: Task1/Valerii-Tcvetkov/linear_classifer.py (hunk 1 above)
Comment (Valerii-Tcvetkov) [reply Task1/Valerii-Tcvetkov/linear_classifer.py]: Внес правки в коммите Edit linear_classifier.py. Fix softmax_with_cross_entropy.
